### PR TITLE
UI: glimmerize masked input

### LINKED ISSUE #6364: UI: Download button on each KV row
**Is your feature request related to a problem? Please describe.**
We are storing SSL-Certificates in a KV pair, we would like to provide a download for those administrators who aren't using shell and apis.

**Describe the solution you'd like**
A Download Button next to "copy button" in a KV for each KV pair

**Describe alternatives you've considered**
-

**Explain any additional use-cases**
-

**Additional context**
-

### PR DESCRIPTION
**Pull request by hashishaw**
This PR glimmerizes the MaskedInput component, and adds an `allowDownload` attribute which adds a download button next to the copy button. As part of this update, the parent component must set the value in the `onChange` callback rather than the component updating the passed argument (which is deprecated behavior)

`allowDownload` has been enabled for KV V2 secret engine data, which fixes #6364

Screenshot 2023-04-28 at 2 46 15 PM

### CODE REVIEW

File: ui/lib/core/addon/components/masked-input.js [hunk 1]
```diff
@@ -3,63 +3,63 @@
...
-  },
-});
+export default class MaskedInputComponent extends Component {
+  textareaId = 'textarea-' + guidFor(this);
+  @tracked showValue = false;
+
+  constructor() {
+    super(...arguments);
+    if (!this.args.onChange && !this.args.displayOnly) {
+      debug('onChange is required for editable Masked Input!');
```
Comment (Monkeychip) [inline ui/lib/core/addon/components/masked-input.js:40]: nice!

File: ui/tests/integration/components/masked-input-test.js [hunk 2]
```diff
@@ -16,68 +16,40 @@ module('Integration | Component | masked input', function (hooks) {
...
-    await component.toggleMasked();
-
-    assert.dom('.masked-value').hasClass('masked-font');
+  test('it renders a download button when allowDownload is true', async function (assert) {
+    await render(hbs`<MaskedInput @allowDownload={{true}} />`);
+    assert.ok(component.downloadButtonIsPresent);
   });
 
   test('it shortens all outputs when displayOnly and masked', async function (assert) {
     this.set('value', '123456789-123456789-123456789');
```
Comment (Monkeychip) [inline ui/tests/integration/components/masked-input-test.js:51]: thank you for updating the tests. an easy miss when you glimmerize a component.

**Review by Monkeychip — APPROVED** [review]
Very very cool. Thank you for tackling. I'll go ahead and update the glimmerize document and assign you to the jira ticket from the glimmer epic.

**Comment by Noobgam** [comment]
Hello folks, for those not super familiar with Vault configuration, is there an option to opt out of this feature?

I've noticed that in our case people just tend to missclick where they expected to see the copy button, so that's some learning curve after upgrading for a feature we're not using at all

**Comment by dhumphries-sainsburys** [comment]
We are also interested in disabling the download button as it downloads the secrets in plaintext and we have a lot of contractors who we can't trust to misplace these plaintext files

**Comment by andyspiers** [comment]
This is an undesirable and surprising feature IMO.

One click and you have downloaded a sensitive value to your local disk unencrypted.

Of course it's _possible_ to view/copy and then save locally but that requires a specific action whereas this is easy to do by accident, almost without noticing.

It would  be great to be able to disable this please.

**Comment by hellobontempo** [comment]
Thank you for the feedback on this. At this time, we do not have a way for operators to disable/enable specific UI features (I have added the request above to our internal planning for UI specific configurations). To more immediately resolve this issue we've added a warning prior to downloading to prevent accidental downloads. This will be released in 1.14.4 and 1.15.0.

**Pull request merged**